cpro/magicollections.py

```diff
@@ -0,0 +1,257 @@
+    BadgeCollection as _BadgeCollection,
+    UserCollection as _UserCollection
+)
+
+from cpro import models, forms, filters, collections_settings, utils
+
+############################################################
+# Account Collection 
+
+class AccountCollection(_AccountCollection):
+    class AddView(_AccountCollection.AddView):
+        after_save            = collections_settings.addAccountAfterSave
+        after_template        = 'include/accountJSstarter'
+        back_to_list_button   = False
+        extra_context         = collections_settings.modAccountExtraContext
+        form_class = collections_settings.getAccountForm
+        # TODO: Not sure how to handle the ENABLED_COLLECTIONS here
+        js_files              = ENABLED_COLLECTIONS['account']['add'].get('js_files', []) + ['mod_account']
+        otherbuttons_template = 'include/advancedButton'
+        redirect_after_add    = collections_settings.redirectAfterAddAccount
+
+    class EditView(_AccountCollection.EditView):
+        after_template = 'include/accountJSstarter'
+        extra_context  = collections_settings.modAccountExtraContext
+        form_class     = forms.AccountFormAdvanced
+        # TODO: Not sure how to handle the ENABLED_COLLECTIONS here
+        js_files       = ENABLED_COLLECTIONS['account']['edit'].get('js_files', []) + ['mod_account']
+
+    class ListView(_AccountCollection.ListView):
+        before_template  = 'include/beforeLeaderboard'
+        distinct         = True
+        default_ordering = '-level'
+        extra_context    = collections_settings.leaderboardExtraContext
+        filter_form      = forms.FilterAccounts
+        # TODO: Replaced by get_queryset
+        filter_queryset  = filters.filterAccounts
+        # TODO: Not sure how to handle the ENABLED_COLLECTIONS here
+        js_files         = ENABLED_COLLECTIONS['account']['list'].get('js_files', []) + ['leaderboard']
+        # TODO: Not sure if this lambda works
+        show_add_button  = lambda request: not request.user.is_authenticated()
+
+############################################################
+# Activity Collection
+
+class ActivityCollection(_ActivityCollection):
+    # TODO: Not sure if this is ported properly since it's directly taken from the old `settings.py` file
+    ACTIVITY_TAGS = [
+        ('cards', _('New Cards')),
+        ('event', _('Event')),
+        ('live', _('Live')),
+        ('comedy', _('Comedy')),
+        ('room', _('Room Decoration')),
+        ('introduction', _('Introduce yourself')),
+        ('idols', _('Idols')),
+        ('anime', _('Anime')),
+        ('cosplay', _('Cosplay')),
+        ('fanart', _('Fan made')),
+        ('merch', _('Merchandise')),
+        ('community', _('Community')),
+        ('unrelated', _('Unrelated')),
+        ('AR Idol Date', 'AR Idol Date'),
+    ]
+
+    # TODO: Not sure if this is ported properly since it's directly taken from the old `settings.py` file
```

Comment:
The logic of this function + the one below should be in the filter form of activities in `forms.py`.